From 88b8742b19cf643cb4d22f09b7a8634eb7930fca Mon Sep 17 00:00:00 2001
From: Eduardo Ferreira <eduardo.barbosa@toradex.com>
Date: Wed, 8 Nov 2023 10:54:46 -0300
Subject: [PATCH] Support docker system config

TorizonCore maintains a system docker configuration which should also be added to the default config search path.

This change was inherited from LmP.

Upstream-Status: Inappropriate [TorizonCore specific]

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
Signed-off-by: Eduardo Ferreira <eduardo.barbosa@toradex.com>
---
 .../github.com/containers/image/v5/pkg/docker/config/config.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vendor/github.com/containers/image/v5/pkg/docker/config/config.go b/vendor/github.com/containers/image/v5/pkg/docker/config/config.go
index 2e79d0ff..ca0a5c5c 100644
--- a/vendor/github.com/containers/image/v5/pkg/docker/config/config.go
+++ b/vendor/github.com/containers/image/v5/pkg/docker/config/config.go
@@ -37,6 +37,7 @@ var (
 	defaultPerUIDPathFormat = filepath.FromSlash("/run/containers/%d/auth.json")
 	xdgConfigHomePath       = filepath.FromSlash("containers/auth.json")
 	xdgRuntimeDirPath       = filepath.FromSlash("containers/auth.json")
+	dockerSystemConfigPath	= filepath.FromSlash("/usr/lib/docker/config.json")
 	dockerHomePath          = filepath.FromSlash(".docker/config.json")
 	dockerLegacyHomePath    = ".dockercfg"
 	nonLinuxAuthFilePath    = filepath.FromSlash(".config/containers/auth.json")
@@ -229,6 +230,8 @@ func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
 			xdgCfgHome = filepath.Join(homeDir, ".config")
 		}
 		paths = append(paths, newAuthPathDefault(filepath.Join(xdgCfgHome, xdgConfigHomePath)))
+		// Load values from system config by default
+		paths = append(paths, newAuthPathDefault(dockerSystemConfigPath))
 		if dockerConfig := os.Getenv("DOCKER_CONFIG"); dockerConfig != "" {
 			paths = append(paths, newAuthPathDefault(filepath.Join(dockerConfig, "config.json")))
 		} else {
-- 
2.34.1

